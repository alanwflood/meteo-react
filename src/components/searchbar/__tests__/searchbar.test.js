import React from "react";
import { render, waitFor } from "@testing-library/react";
import SearchBar from "../searchbar";
jest.mock(
  "@googlemaps/js-api-loader",
  () =>
    require.requireActual("../../../../tests/googleMapsApiLoaderMock").default
);
jest.mock(
  "react-places-autocomplete",
  () =>
    require.requireActual("../../../../tests/reactPlacesAutocompleteMock")
      .default
);

describe("SearchBar component", () => {
  const onSubmit = jest.fn();
  const address = "Dublin, Ireland";

  beforeEach(() => {
    window.google = {};
  });

  it("Renders correctly", () => {
    const { container } = render(
      <SearchBar onSubmit={onSubmit} address={address} />
    );
    expect(container).toMatchSnapshot();
  });

  it("Loads Google Maps Api on mount", async () => {
    window.google = undefined;
    render(<SearchBar onSubmit={onSubmit} address={address} />);
    await waitFor(() => {
      expect(window.google).toBe("Google Maps Loaded");
    });
  });
});
