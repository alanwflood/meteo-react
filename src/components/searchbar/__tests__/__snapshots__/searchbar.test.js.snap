// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component Renders correctly 1`] = `
<div>
  <form
    class="search"
  >
    <div
      class="search-bar"
    >
      <input
        aria-labelledby="searchButton"
        class="search-input"
        placeholder="Search Places ..."
      />
      <div
        class="search-dropdown"
      />
    </div>
    <button
      aria-label="Search for a location to Fetch weather for"
      id="searchButton"
      type="submit"
    >
      <span>
        Search
      </span>
      <test-file-stub />
    </button>
  </form>
</div>
`;
